features/environment.py

```diff
@@ -275,6 +277,14 @@ def __init__(
             job_suffix = job_suffix.split("PR-")[-1]
         timed_job_tag += str(job_suffix)
         timed_job_tag = timed_job_tag.replace(".", "-")
+
+        # Add 8-digit random suffix to the tag, so it does not conflict when
+        # spinning many jobs at the same time
+        random_suffix = "".join(
+            random.choices(string.ascii_lowercase + string.digits, k=8)
+        )
+        timed_job_tag += random_suffix
```

Comment:
Currently the name ends up like this: `i0-uaclient-ci-0830-1427-dev8e1jf5s9`
It'd be nicer if there was a dash before the random suffix, like this: `i0-uaclient-ci-0830-1427-dev-8e1jf5s9`.
```suggestion
        timed_job_tag += "-" + random_suffix
```

Reply:
Done! Looks better now.